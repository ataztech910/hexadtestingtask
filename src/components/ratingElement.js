import React from 'react';
import IconButton from '@material-ui/core/IconButton';
import Icon from '@material-ui/core/Icon';
import ListItemAvatar from '@material-ui/core/ListItemAvatar';
import Avatar from '@material-ui/core/Avatar';
import ListItemText from '@material-ui/core/ListItemText';
import ListItemSecondaryAction from '@material-ui/core/ListItemSecondaryAction';

export const RatingElement = props => (
  <>
    <ListItemAvatar>
      <Avatar
        alt={`Image for Â°${props.name}`}
        src={`https://picsum.photos/80/80?image=${props.imgId}`}
      />
    </ListItemAvatar>
    <ListItemText className="elementRating" primary={props.name} secondary={`Rating is: ${props.rating}`} />
    <ListItemSecondaryAction>
      <IconButton
        color="primary"
        component="span"
        className="increaseRating"
        onClick={(event) => {
          event.preventDefault();
          props.setRating({ id: props.id, action: 1 });
        }}
      >
        <Icon>thumb_up</Icon>
      </IconButton>
      <IconButton
        color="primary"
        className="decreaseRating"
        component="span"
        onClick={(event) => {
          event.preventDefault();
          props.setRating({ id: props.id, action: -1 });
        }}
      >
        <Icon>thumb_down</Icon>
      </IconButton>
    </ListItemSecondaryAction>
  </>
);
export default RatingElement;
