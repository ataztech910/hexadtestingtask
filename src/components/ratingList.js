import React from 'react';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import RatingElementContainer from '../containers/RatingElementConatainer';

export const RatingList = list => (
  <List className="rating-list">
    {list.fetch.list.ratings && list.fetch.list.ratings
      ? list.fetch.list.ratings.map(element => <ListItem alignItems="flex-start" className="element" key={`element-${element.id}`}><RatingElementContainer {...element} /></ListItem>)
      : 'No elements, pitty =('}
  </List>
);
export default RatingList;
