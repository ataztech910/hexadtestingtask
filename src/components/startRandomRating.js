import React from 'react';
import Button from '@material-ui/core/Button';

const StartRandomRating = props => (
  <Button
    variant="contained"
    color="primary"
    type="button"
    onClick={(event) => {
      event.preventDefault();
      props.startRandomProcess();
    }}
  >
    {props.state.startProcess ? 'Stop random rates' : 'Start random rates' }
  </Button>
);

export default StartRandomRating;
