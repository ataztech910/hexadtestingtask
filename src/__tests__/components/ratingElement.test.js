import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import { RatingElement } from '../../components/ratingElement';

configure({ adapter: new Adapter() });

describe('Rating element check', () => {
  const props = { name: 'Test', rating: 3 };
  const element = shallow(<RatingElement {...props} />);
  it('renders properly', () => {
    expect(element).toMatchSnapshot();
  });

  it('displays the rating from props', () => {
    expect(element.find('.elementRating').props()['secondary']).toEqual('Rating is: 3');
  });
});
