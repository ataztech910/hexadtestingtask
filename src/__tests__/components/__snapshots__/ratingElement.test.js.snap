// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating element check renders properly 1`] = `ShallowWrapper {}`;
